src/resumable-subscription.ts

```diff
@@ -1,130 +1,102 @@
+        
+        this.options = replaceUnimplementedListenersWithNoOps(options);
     }
-
-    tryNow(): void {
-        this.state = ResumableSubscriptionState.OPENING;
+    
+    private tryNow(): void {
         let newXhr = this.xhrSource();
-        this.subscription = new Subscription(newXhr, {
-            path: this.options.path,
-            lastEventId: this.lastEventIdReceived,
-            onOpen: () => {
-                this.assertState(['OPENING']);
-                this.state = ResumableSubscriptionState.OPEN;
-                if (this.options.onOpen) { this.options.onOpen(); }
-                this.retryStrategy.reset(); //We need to reset the counter once the connection has been re-established.
-            },
-            onEvent: (event: Event) => {
-                this.assertState(['OPEN']);
-                if (this.options.onEvent) { this.options.onEvent(event); }
-                console.assert(
-                    !this.lastEventIdReceived ||
-                    parseInt(event.eventId) > parseInt(this.lastEventIdReceived),
-                    'Expected the current event id to be larger than the previous one'
-                );
-                this.lastEventIdReceived = event.eventId;
-            },
-            onEnd: () => {
-                this.state = ResumableSubscriptionState.ENDED;
-                if (this.options.onEnd) { this.options.onEnd(); }
-            },
-            onError: (error: Error) => {
-                this.state = ResumableSubscriptionState.OPENING
-                this.retryStrategy.attemptRetry(error)
-                .then(() => {
-                  if (this.options.onRetry !== undefined) {
-                    this.options.onRetry();
-                  } else {
-                    this.tryNow();
-                  }
-                })
-                .catch(error => {
-                    this.state = ResumableSubscriptionState.ENDED;
-                    if (this.options.onError) { this.options.onError(error); }
-                })},
-            logger: this.logger
-        });
-               
-        if (this.options.tokenProvider) {
-            this.options.tokenProvider.fetchToken().then((jwt) => {
-                this.subscription.open(jwt);
-            }).catch((err) => {
-                if(this.options.onError) this.options.onError(err);
+        
+        if (this.lastEventIdReceived) {
+            newXhr.setRequestHeader("Last-Event-Id", this.lastEventIdReceived);
+        }
+        
+        this.options.tokenProvider.fetchToken()
+        .then( token => {
+            this.baseSubscription = new BaseSubscription(newXhr, {
+                path: this.options.path,
+                headers: this.options.headers,
+                jwt: token,
+                
+                onOpen: () => {
+                    this.options.onOpen();
+                    this.retryStrategy.reset(); //We need to reset the counter once the connection has been re-established.
```

Comment:
I also agree that that's a good idea, but couldn't design this in a nice and generic enough way for subscriptions and requests to work together. I think it'll have to make do for now.